plugins/com.act21.mamba.layout.alg.tree/src/com/act21/mamba/layout/alg/tree/TreeLayoutProvider.java

```diff
@@ -0,0 +1,210 @@
+package com.act21.mamba.layout.alg.tree;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.eclipse.elk.core.AbstractLayoutProvider;
+import org.eclipse.elk.core.util.IElkProgressMonitor;
+import org.eclipse.elk.graph.ElkBendPoint;
+import org.eclipse.elk.graph.ElkEdge;
+import org.eclipse.elk.graph.ElkEdgeSection;
+import org.eclipse.elk.graph.ElkGraphFactory;
+import org.eclipse.elk.graph.ElkNode;
+import org.eclipse.emf.common.util.EList;
+
+/**
+ * A simple layout algorithm class. This algorithm already supports a number of
+ * layout options, places nodes, and routes edges.....
+ */
+public class TreeLayoutProvider extends AbstractLayoutProvider {
+
+	@Override
+	public void layout(ElkNode layoutGraph, IElkProgressMonitor progressMonitor) {
```

Comment:
You might want to consider t use the progressMonitor to log graphs after certain parts of you algorithm